src/iden3comm/handlers/credential-proposal.ts

```diff
@@ -218,16 +219,37 @@ export class CredentialProposalHandler
 
     let credOfferMessage: CredentialsOfferMessage | undefined = undefined;
     let proposalMessage: ProposalMessage | undefined = undefined;
+
+    let isExternalSender = false;
+
+    try {
+      await this._identityWallet.getGenesisDIDMetadata(senderDID);
+    } catch {
+      isExternalSender = true;
+    }
     for (let i = 0; i < proposalRequest.body.credentials.length; i++) {
       const cred = proposalRequest.body.credentials[i];
 
       // check if there is credentials in the wallet
       let credsFromWallet: W3CCredential[] = [];
+
       try {
-        credsFromWallet = await this._identityWallet.findOwnedCredentialsByDID(senderDID, {
-          type: cred.type,
-          context: cred.context
-        });
+        if (isExternalSender) {
+          credsFromWallet = await this._credWallet.findByQuery({
```

Comment:
why can't we always use this? if it is 'internal' sender I think it work in the same way.